### PR TITLE
fixes #2281 - don't allow admin flag to be removed from admin account

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

**Comment by domcleal** [comment]
[test]

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -59,11 +59,14 @@ def update
     if @user.update_attributes(params[:user])
       # Only an admin can update admin attribute of another use
       # this is required, as the admin field is blacklisted above
-      @user.update_attribute(:admin, admin) if User.current.admin
-      @user.roles << Role.find_by_name("Anonymous") unless @user.roles.map(&:name).include? "Anonymous"
-      hostgroup_ids = params[:user]["hostgroup_ids"].reject(&:empty?).map(&:to_i) unless params[:user]["hostgroup_ids"].empty?
-      update_hostgroups_owners(hostgroup_ids) unless hostgroup_ids.empty?
-      process_success editing_self ? { :success_redirect => hosts_path } : {}
+      if User.current.admin && (@user.admin = admin; !@user.save)
```
Comment (ohadlevy) [inline app/controllers/users_controller.rb]: would it be enough to do @user.valid?

``` ruby
if User.current.admin? 
  @user.admin = admin
  return process_error unless @user.valid?
end
```

Comment (domcleal) [inline app/controllers/users_controller.rb]: Thanks, that's neater when done before the update_attributes.  Branch updated + rebased.

**Comment by lzap** [comment]
I think the patch does not count with good scenario in new user action.

Fixed here: https://github.com/theforeman/foreman/pull/583

**Comment by domcleal** [comment]
@lzap hm?  This is a different bug to the mass assignment w/admin flag when creating users.

**Comment by lzap** [comment]
Oh yes, älso I removed it from the patch. There is other PR with the proper fix. sorry. https://github.com/theforeman/foreman/pull/563

**Comment by ohadlevy** [comment]
rebase please?

**Comment by domcleal** [comment]
[test]

**Comment by ohadlevy** [comment]
merged - thanks

**Pull request closed without merging**